# Analisis subset C - Posiciones Corporales

# 0. Descripción de los subsets del grupo C

Este subset decribe la posición corporal de las personas involucradas en el experimento.

+ C - Body posture features (Kinect - final (annotated and selected))-sheet_1 y C - Body posture features (Kinect - final (annotated and selected))-sheet_4 :
En este archivo se recopilan las features que se han generado a través de del reconocimiento de las posturas corporales de los sujetos en el experimento. Se usa el sensor Kinect 3D, extrayendo las features con Kinect SDK. Hay varias columnas que se detallan a continuación:

  *Metadatos generales:*
  - **PP**: Identificador del participante.
  - **C**: Condición experimental en la que se realizó la medición.
  - **timestamp**: Marca de tiempo indicando el minuto exacto de la medición.

  *Distancia:*
  - **avgDepth(avg y stdv)**: Profundidad promedio del participante en la escena, medida por el sensor.

  *Ángulos articulares (Joint Angles):*
  - **leftShoulderAngle(avg y stdv)**: Ángulo promedio del hombro izquierdo.
  - **rightShoulderAngle(avg y stdv)**: Ángulo promedio del hombro derecho.
  - **leanAngle(avg y stdv)**: Ángulo de inclinación promedio del torso.
  - **HipCenter_Spine-Spine_ShoulderCenter(avg y stdv)**: Ángulo entre la cadera y la parte superior de la columna.
  - **Spine_ShoulderCenter-ShoulderCenter_Head(avg y stdv)**: Ángulo entre la parte superior de la columna y la cabeza.
  - **Spine_ShoulderCenter-ShoulderCenter_ShoulderLeft(avg y stdv)**: Ángulo entre la parte superior de la columna y el hombro izquierdo.
  - **Spine_ShoulderCenter-ShoulderCenter_ShoulderRight(avg y stdv)**: Ángulo entre la parte superior de la columna y el hombro derecho.
  - **ShoulderCenter_ShoulderLeft-ShoulderLeft_ElbowLeft(avg y stdv)**: Ángulo entre el hombro izquierdo y el codo izquierdo.
  - **ShoulderLeft_ElbowLeft-ElbowLeft_WristLeft(avg y stdv)**: Ángulo entre el codo izquierdo y la muñeca izquierda.
  - **ShoulderCenter_ShoulderRight-ShoulderRight_ElbowRight(avg y stdv)**: Ángulo entre el hombro derecho y el codo derecho.
  - **ElbowLeft_WristLeft-WristLeft_HandLeft(avg y stdv)**: Ángulo entre el codo izquierdo, la muñeca izquierda y la mano izquierda.
  - **ShoulderCenter_ShoulderRight-ShoulderRight_ElbowRight(avg y stdv)**: Ángulo entre el centro de los hombros, el hombro derecho y el codo derecho.
  - **ShoulderRight_ElbowRight-ElbowRight_WristRight(avg y stdv)**: Ángulo entre el hombro derecho, el codo derecho y la muñeca derecha.
  - **ElbowRight_WristRight-WristRight_HandRight(avg y stdv)**: Ángulo entre el codo derecho, la muñeca derecha y la mano derecha.

  *Orientaciones óseas (Bone Orientations) en tres planos (ZX, XY, YZ):*
  - **HipCenter_Spine-PlaneZXAxisX(avg y stdv)**: Orientación del hueso entre la cadera y la columna en el plano ZX.
  - **HipCenter_Spine-PlaneXYAxisY(avg y stdv)**: Orientación del hueso entre la cadera y la columna en el plano XY.
  - **HipCenter_Spine-PlaneYZAxisZ(avg y stdv)**: Orientación del hueso entre la cadera y la columna en el plano YZ.
  - **Spine_ShoulderCenter-PlaneZXAxisX(avg y stdv)**: Orientación del hueso entre la parte superior de la columna y el centro de los hombros en el plano ZX.
  - **Spine_ShoulderCenter-PlaneXYAxisY(avg y stdv)**: Orientación del hueso entre la parte superior de la columna y el centro de los hombros en el plano XY.
  - **Spine_ShoulderCenter-PlaneYZAxisZ(avg y stdv)**: Orientación del hueso entre la parte superior de la columna y el centro de los hombros en el plano YZ.
  - **ShoulderCenter_Head-PlaneZXAxisX(avg y stdv)**: Orientación del hueso entre el centro de los hombros y la cabeza en el plano ZX.
  - **ShoulderCenter_Head-PlaneXYAxisY(avg y stdv)**: Orientación del hueso entre el centro de los hombros y la cabeza en el plano XY.
  - **ShoulderCenter_Head-PlaneYZAxisZ(avg y stdv)**: Orientación del hueso entre el centro de los hombros y la cabeza en el plano YZ.
  - **ShoulderCenter_ShoulderLeft-PlaneZXAxisX(avg y stdv)**: Orientación del hueso entre el centro de los hombros y el hombro izquierdo en el plano ZX.
  - **ShoulderCenter_ShoulderLeft-PlaneXYAxisY(avg y stdv)**: Orientación del hueso entre el centro de los hombros y el hombro izquierdo en el plano XY.
  - **ShoulderCenter_ShoulderLeft-PlaneYZAxisZ(avg y stdv)**: Orientación del hueso entre el centro de los hombros y el hombro izquierdo en el plano YZ.
  - **ShoulderCenter_ShoulderRight-PlaneZXAxisX(avg y stdv)**: Orientación del hueso entre el centro de los hombros y el hombro derecho en el plano ZX.
  - **ShoulderCenter_ShoulderRight-PlaneXYAxisY(avg y stdv)**: Orientación del hueso entre el centro de los hombros y el hombro derecho en el plano XY.
  - **ShoulderCenter_ShoulderRight-PlaneYZAxisZ(avg y stdv)**: Orientación del hueso entre el centro de los hombros y el hombro derecho en el plano YZ.
  - **ShoulderLeft_ElbowLeft-PlaneZXAxisX(avg y stdv)**: Orientación del hueso entre el hombro izquierdo y el codo izquierdo en el plano ZX.
  - **ShoulderLeft_ElbowLeft-PlaneXYAxisY(avg y stdv)**: Orientación del hueso entre el hombro izquierdo y el codo izquierdo en el plano XY.
  - **ShoulderLeft_ElbowLeft-PlaneYZAxisZ(avg y stdv)**: Orientación del hueso entre el hombro izquierdo y el codo izquierdo en el plano YZ.
  - **ElbowLeft_WristLeft-PlaneZXAxisX(avg y stdv)**: Orientación del hueso entre el codo izquierdo y la muñeca izquierda en el plano ZX.
  - **ElbowLeft_WristLeft-PlaneXYAxisY(avg y stdv)**: Orientación del hueso entre el codo izquierdo y la muñeca izquierda en el plano XY.
  - **ElbowLeft_WristLeft-PlaneYZAxisZ(avg y stdv)**: Orientación del hueso entre el codo izquierdo y la muñeca izquierda en el plano YZ.
  - **WristLeft_HandLeft-PlaneZXAxisX(avg y stdv)**: Orientación del hueso entre la muñeca izquierda y la mano izquierda en el plano ZX.
  - **WristLeft_HandLeft-PlaneXYAxisY(avg y stdv)**: Orientación del hueso entre la muñeca izquierda y la mano izquierda en el plano XY.
  - **WristLeft_HandLeft-PlaneYZAxisZ(avg y stdv)**: Orientación del hueso entre la muñeca izquierda y la mano izquierda en el plano YZ.
  - **ShoulderRight_ElbowRight-PlaneZXAxisX(avg y stdv)**: Orientación del hueso entre el hombro derecho y el codo derecho en el plano ZX.
  - **ShoulderRight_ElbowRight-PlaneXYAxisY(avg y stdv)**: Orientación del hueso entre el hombro derecho y el codo derecho en el plano XY.
  - **ShoulderRight_ElbowRight-PlaneYZAxisZ(avg y stdv)**: Orientación del hueso entre el hombro derecho y el codo derecho en el plano YZ.
  - **ElbowRight_WristRight-PlaneZXAxisX(avg y stdv)**: Orientación del hueso entre el codo derecho y la muñeca derecha en el plano ZX.
  - **ElbowRight_WristRight-PlaneXYAxisY(avg y stdv)**: Orientación del hueso entre el codo derecho y la muñeca derecha en el plano XY.
  - **ElbowRight_WristRight-PlaneYZAxisZ(avg y stdv)**: Orientación del hueso entre el codo derecho y la muñeca derecha en el plano YZ.
  - **WristRight_HandRight-PlaneZXAxisX(avg y stdv)**: Orientación del hueso entre la muñeca derecha y la mano derecha en el plano ZX.
  - **WristRight_HandRight-PlaneXYAxisY(avg y stdv)**: Orientación del hueso entre la muñeca derecha y la mano derecha en el plano XY.
  - **WristRight_HandRight-KinectZAxis(avg y stdv)**: Orientación del hueso entre la muñeca derecha y la mano derecha en el eje KinectZ.


+ C - Body posture features (Kinect - final (annotated and selected))-sheet_2:
Es todo igual menos los metadatos generales

  *Metadatos generales:*
  - **PP**: Identificador del participante.
  - **Blok**: Bloque experimental en el que se encuentra el participante.
  - **Condition**: Condición experimental en la que se realizó la medición.
  - **timestamp**: Marca de tiempo indicando el minuto exacto de la medición.

+ C - Body posture features (Kinect - final (annotated and selected))-sheet_3:
Este sólo contiene el nombre de las columnas

In [ ]:
from utils import *

In [ ]:
sheet_1 = cargar_datos_csv("C - Body posture features (Kinect - final (annotated and selected))-sheet_1.csv")
sheet_2 = cargar_datos_csv("C - Body posture features (Kinect - final (annotated and selected))-sheet_2.csv")
sheet_4 = cargar_datos_csv("C - Body posture features (Kinect - final (annotated and selected))-sheet_4.csv")

Datos cargados correctamente: C - Body posture features (Kinect - final (annotated and selected))-sheet_1.csv
Datos cargados correctamente: C - Body posture features (Kinect - final (annotated and selected))-sheet_2.csv
Datos cargados correctamente: C - Body posture features (Kinect - final (annotated and selected))-sheet_4.csv


In [ ]:
sheet_1.shape, sheet_2.shape, sheet_4.shape

((3304, 97), (3139, 99), (3235, 98))

In [ ]:
sheet_1.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3304 entries, 0 to 3303
Data columns (total 97 columns):
 #   Column                                                       Non-Null Count  Dtype 
---  ------                                                       --------------  ----- 
 0   PP                                                           3304 non-null   object
 1   C                                                            3304 non-null   object
 2   timestamp                                                    3304 non-null   object
 3   avgDepth(avg)                                                3304 non-null   object
 4   leftShoulderAngle(avg)                                       3304 non-null   object
 5   rightShoulderAngle(avg)                                      3291 non-null   object
 6   leanAngle(avg)                                               3291 non-null   object
 7   HipCenter_Spine-Spine_ShoulderCenter(avg)                    3291 non-null   object
 8 

In [ ]:
sheet_2.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3139 entries, 0 to 3138
Data columns (total 99 columns):
 #   Column                                                       Non-Null Count  Dtype  
---  ------                                                       --------------  -----  
 0   PP                                                           3139 non-null   object 
 1   Blok                                                         3139 non-null   int64  
 2   Condition                                                    3139 non-null   object 
 3   timestamp                                                    3139 non-null   object 
 4   avgDepth(avg)                                                2741 non-null   float64
 5   leftShoulderAngle(avg)                                       2741 non-null   float64
 6   rightShoulderAngle(avg)                                      2738 non-null   float64
 7   leanAngle(avg)                                               2738 non-null   f

In [ ]:
sheet_2["Added"]

0       NaN
1       NaN
2       NaN
3       NaN
4       NaN
       ... 
3134    NaN
3135    NaN
3136    NaN
3137    NaN
3138    NaN
Name: Added, Length: 3139, dtype: object

In [ ]:
sheet_4.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3235 entries, 0 to 3234
Data columns (total 98 columns):
 #   Column                                                       Non-Null Count  Dtype  
---  ------                                                       --------------  -----  
 0   PP                                                           3235 non-null   object 
 1   C                                                            3235 non-null   int64  
 2   Unnamed: 2                                                   0 non-null      float64
 3   timestamp                                                    3235 non-null   object 
 4   avgDepth(avg)                                                3235 non-null   float64
 5   leftShoulderAngle(avg)                                       3235 non-null   float64
 6   rightShoulderAngle(avg)                                      3222 non-null   float64
 7   leanAngle(avg)                                               3222 non-null   f

No hace falta seguir el análisis para comparar los diferentes archivos, vemos como sheet_1 y sheet_4 tiene 3106 datos de fecha pero tiene más datos de PP, por lo que hay experimentos sin medción de tiempo y no cumple con la serie temporal demostrada en otros datasets.
Sheet_2 sin embargo cumple con la serie temporal de 3139 registros.

In [ ]:
formatear_fechas(sheet_2)

PP  Blok Condition           timestamp  avgDepth(avg)  \
0      PP1     1         R 2012-08-18 13:16:00    2102.597393   
1      PP1     1         R 2012-08-18 13:17:00    2099.725525   
2      PP1     1         R 2012-08-18 13:18:00    2102.365778   
3      PP1     1         R 2012-08-18 13:19:00    2104.116968   
4      PP1     1         R 2012-08-18 13:20:00    2105.284007   
...    ...   ...       ...                 ...            ...   
3134  PP25     3         T 2012-10-07 16:15:00    2047.087447   
3135  PP25     3         T 2012-10-07 16:16:00    2048.832532   
3136  PP25     3         T 2012-10-07 16:17:00    2048.863806   
3137  PP25     3         T 2012-10-07 16:18:00    2049.784363   
3138  PP25     3         T 2012-10-07 16:19:00    2048.695384   

      leftShoulderAngle(avg)  rightShoulderAngle(avg)  leanAngle(avg)  \
0                -116.055931               115.017758       92.340895   
1                -116.301605               115.986636       92.083385   
2                -115.963089               114.073054       92.381690   
3                -115.629630               113.972465       92.428905   
4                -116.359699               111.538437       92.461537   
...                      ...                      ...             ...   
3134             -115.219804               116.330627      104.529078   
3135             -115.124481               115.231220      102.381525   
3136             -115.138635               113.786573      103.612981   
3137             -114.085126               109.726265      108.599482   
3138             -114.335215               112.478682      104.207235   

      HipCenter_Spine-Spine_ShoulderCenter(avg)  \
0                                    176.389544   
1                                    178.236949   
2                                    170.876998   
3                                    168.248852   
4                                    166.050980   
...                                         ...   
3134                                 176.843390   
3135                                 176.970969   
3136                                 176.990190   
3137                                 176.763443   
3138                                 177.046208   

      Spine_ShoulderCenter-ShoulderCenter_Head(avg)  ...  \
0                                        155.409463  ...   
1                                        155.185113  ...   
2                                        155.363873  ...   
3                                        156.627814  ...   
4                                        152.687506  ...   
...                                             ...  ...   
3134                                     160.731168  ...   
3135                                     154.387065  ...   
3136                                     156.480959  ...   
3137                                     159.023415  ...   
3138                                     159.742007  ...   

      ShoulderRight_ElbowRight-PlaneZXAxisX(stdv)  \
0                                       22.946337   
1                                        7.194860   
2                                       14.726365   
3                                        2.542686   
4                                        9.905241   
...                                           ...   
3134                                    36.355940   
3135                                    14.714479   
3136                                    22.742133   
3137                                    28.595668   
3138                                    24.310952   

      ShoulderRight_ElbowRight-PlaneXYAxisY(stdv)  \
0                                        5.605230   
1                                        2.468881   
2                                        2.953021   
3                                        1.564318   
4                                       20.342488   
...                                           ...   
3134    

In [ ]:
#Conertir Blok a object
sheet_2["Blok"] = sheet_2["Blok"].astype(str)

In [ ]:
#Elimina columnas en sheet_2
sheet_2 = eliminar_columnas(sheet_2,["Added"])

Se eliminaron las columnas: ['Added']


In [ ]:
pd.set_option('display.max_rows', None)
evaluar_datos_faltantes(sheet_2)

Columna  \
ShoulderCenter_ShoulderLeft-PlaneZXAxisX(stdv)         ShoulderCenter_ShoulderLeft-PlaneZXAxisX(stdv)   
ShoulderLeft_ElbowLeft-PlaneYZAxisZ(stdv)                   ShoulderLeft_ElbowLeft-PlaneYZAxisZ(stdv)   
leftShoulderAngle(stdv)                                                       leftShoulderAngle(stdv)   
ShoulderCenter_ShoulderLeft-PlaneXYAxisY(stdv)         ShoulderCenter_ShoulderLeft-PlaneXYAxisY(stdv)   
ShoulderCenter_ShoulderLeft-PlaneYZAxisZ(stdv)         ShoulderCenter_ShoulderLeft-PlaneYZAxisZ(stdv)   
ShoulderLeft_ElbowLeft-PlaneXYAxisY(stdv)                   ShoulderLeft_ElbowLeft-PlaneXYAxisY(stdv)   
ElbowLeft_WristLeft-PlaneYZAxisZ(stdv)                         ElbowLeft_WristLeft-PlaneYZAxisZ(stdv)   
WristLeft_HandLeft-PlaneYZAxisZ(stdv)                           WristLeft_HandLeft-PlaneYZAxisZ(stdv)   
ShoulderRight_ElbowRight-PlaneYZAxisZ(stdv)               ShoulderRight_ElbowRight-PlaneYZAxisZ(stdv)   
ShoulderRight_ElbowRight-PlaneZXAxisX(stdv)               ShoulderRight_ElbowRight-PlaneZXAxisX(stdv)   
WristLeft_HandLeft-PlaneZXAxisX(stdv)                           WristLeft_HandLeft-PlaneZXAxisX(stdv)   
ShoulderLeft_ElbowLeft-PlaneZXAxisX(stdv)                   ShoulderLeft_ElbowLeft-PlaneZXAxisX(stdv)   
ElbowLeft_WristLeft-PlaneZXAxisX(stdv)                         ElbowLeft_WristLeft-PlaneZXAxisX(stdv)   
ShoulderCenter_ShoulderRight-PlaneYZAxisZ(stdv)       ShoulderCenter_ShoulderRight-PlaneYZAxisZ(stdv)   
ElbowRight_WristRight-PlaneZXAxisX(stdv)                     ElbowRight_WristRight-PlaneZXAxisX(stdv)   
WristRight_HandRight-KinectZAxis(stdv)                         WristRight_HandRight-KinectZAxis(stdv)   
WristRight_HandRight-PlaneXYAxisY(stdv)                       WristRight_HandRight-PlaneXYAxisY(stdv)   
ElbowRight_WristRight-PlaneXYAxisY(stdv)                     ElbowRight_WristRight-PlaneXYAxisY(stdv)   
ShoulderCenter_Head-PlaneZXAxisX(stdv)                         ShoulderCenter_Head-PlaneZXAxisX(stdv)   
HipCenter_Spine-PlaneZXAxisX(stdv)                                 HipCenter_Spine-PlaneZXAxisX(stdv)   
Spine_ShoulderCenter-PlaneZXAxisX(stdv)                       Spine_ShoulderCenter-PlaneZXAxisX(stdv)   
Spine_ShoulderCenter-PlaneXYAxisY(stdv)                       Spine_ShoulderCenter-PlaneXYAxisY(stdv)   
WristRight_HandRight-PlaneZXAxisX(stdv)                       WristRight_HandRight-PlaneZXAxisX(stdv)   
ElbowLeft_WristLeft-PlaneXYAxisY(stdv)                         ElbowLeft_WristLeft-PlaneXYAxisY(stdv)   
HipCenter_Spine-PlaneXYAxisY(stdv)                                 HipCenter_Spine-PlaneXYAxisY(stdv)   
ElbowRight_WristRight-PlaneYZAxisZ(stdv)                     ElbowRight_WristRight-PlaneYZAxisZ(stdv)   
WristLeft_HandLeft-PlaneXYAxisY(stdv)                           WristLeft_HandLeft-PlaneXYAxisY(stdv)   
ShoulderCenter_Head-PlaneYZAxisZ(stdv)                         ShoulderCenter_Head-PlaneYZAxisZ(stdv)   
ShoulderLeft_ElbowLeft-ElbowLeft_WristLeft(stdv)     ShoulderLeft_ElbowLeft-ElbowLeft_WristLeft(stdv)   
ShoulderRight_ElbowRight-PlaneXYAxisY(stdv)               ShoulderRight_ElbowRight-PlaneXYAxisY(stdv)   
rightShoulderAngle(stdv)                                                     rightShoulderAngle(stdv)   
leanAngle(stdv)                                                                       leanAngle(stdv)   
HipCenter_Spine-Spine_ShoulderCenter(stdv)                 HipCenter_Spine-Spine_ShoulderCenter(stdv)   
Spine_ShoulderCenter-ShoulderCenter_Head(stdv)         Spine_ShoulderCenter-ShoulderCenter_Head(stdv)   
Spine_ShoulderCenter-ShoulderCenter_ShoulderLef...  Spine_ShoulderCenter-ShoulderCenter_ShoulderLe...   
Spine_ShoulderCenter-ShoulderCenter_ShoulderRig...  Spine_ShoulderCenter-ShoulderCenter_ShoulderRi...   
ShoulderCenter_ShoulderLeft-ShoulderLeft_ElbowL...  ShoulderCenter_ShoulderLeft-ShoulderLeft_Elbow...   
ShoulderCenter_ShoulderRight-ShoulderRight_Elbo...  ShoulderCenter_ShoulderRight-ShoulderRight_Elb...   

In [ ]:
#auto_na_imputer(sheet_2)